lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -0,0 +1,472 @@
+	Results []EnrollEKSClusterResult
+}
+
+// EnrollEKSCLusterClient defines functions required for EKS cluster enrollment.
+type EnrollEKSCLusterClient interface {
+	// CreateAccessEntry creates an access entry. An access entry allows an IAM principal to access an EKS cluster.
+	CreateAccessEntry(ctx context.Context, params *eks.CreateAccessEntryInput, optFns ...func(*eks.Options)) (*eks.CreateAccessEntryOutput, error)
+
+	// AssociateAccessPolicy associates an access policy and its scope to an access entry.
+	AssociateAccessPolicy(ctx context.Context, params *eks.AssociateAccessPolicyInput, optFns ...func(*eks.Options)) (*eks.AssociateAccessPolicyOutput, error)
+
+	// ListAccessEntries lists the access entries for an EKS cluster.
+	ListAccessEntries(ctx context.Context, params *eks.ListAccessEntriesInput, optFns ...func(*eks.Options)) (*eks.ListAccessEntriesOutput, error)
+
+	// DeleteAccessEntry deletes an access entry from an EKS cluster.
+	DeleteAccessEntry(ctx context.Context, params *eks.DeleteAccessEntryInput, optFns ...func(*eks.Options)) (*eks.DeleteAccessEntryOutput, error)
+
+	// DescribeCluster returns detailed information about an EKS cluster.
+	DescribeCluster(ctx context.Context, params *eks.DescribeClusterInput, optFns ...func(*eks.Options)) (*eks.DescribeClusterOutput, error)
+
+	// GetCallerIdentity returns details about the IAM user or role whose credentials are used to call the operation.
+	GetCallerIdentity(ctx context.Context, params *sts.GetCallerIdentityInput, optFns ...func(*sts.Options)) (*sts.GetCallerIdentityOutput, error)
+
+	// CheckAgentAlreadyInstalled checks if teleport-kube-agent Helm chart is already installed on the EKS cluster.
+	CheckAgentAlreadyInstalled(genericclioptions.RESTClientGetter, logrus.FieldLogger) (bool, error)
+
+	// InstallKubeAgent installs teleport-kube-agent Helm chart to the EKS cluster.
+	InstallKubeAgent(context.Context, *eksTypes.Cluster, string, genericclioptions.RESTClientGetter, logrus.FieldLogger, EnrollEKSClustersRequest) error
+}
+
+type defaultEnrollEKSClustersClient struct {
+	*eks.Client
+	stsClient *sts.Client
+}
+
+func (d *defaultEnrollEKSClustersClient) GetCallerIdentity(ctx context.Context, params *sts.GetCallerIdentityInput, optFns ...func(*sts.Options)) (*sts.GetCallerIdentityOutput, error) {
+	return d.stsClient.GetCallerIdentity(ctx, params, optFns...)
+}
+
+func (d *defaultEnrollEKSClustersClient) CheckAgentAlreadyInstalled(kubeconfig genericclioptions.RESTClientGetter, log logrus.FieldLogger) (bool, error) {
+	actionConfig, err := getHelmActionConfig(kubeconfig, log)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	return checkAgentAlreadyInstalled(actionConfig)
+}
+
+func (d *defaultEnrollEKSClustersClient) InstallKubeAgent(ctx context.Context, eksCluster *eksTypes.Cluster, proxyAddr string, kubeconfig genericclioptions.RESTClientGetter, log logrus.FieldLogger, req EnrollEKSClustersRequest) error {
```

Comment:
Don't name it kubeconfig as it might create the impression that's a kubeconfig file